### PR TITLE
APP-3058 Activities in SpringBoot Starter

### PR DESCRIPTION
**Pull request by symphony-thibault**
### Ticket
[APP-3058](https://perzoinc.atlassian.net/browse/APP-3058)

### Description
Ported Activity API to Core Starter.

#### Slash
Registering a new Slash command in SpringBoot context looks like:
```java
@Component
public class SlashBean {

  @Slash("/test")
  public void onTest(CommandContext context) {

  }
}
```

#### Real Time Events
Subscribing to a real time event is now as simple as:
```java
@Component
public class RealTimeEventsTestListener {

  @EventListener
  public void onMessageSent(RealTimeEvent<V4MessageSent> event) {

  }

  @EventListener
  public void onSharedPost(RealTimeEvent<V4SharedPost> event) {

  }
}
``` 

### Checklist
- [x] Referenced a ticket in the PR title and in the corresponding section
- [ ] Filled properly the description and dependencies, if any
- [x] Unit tests updated or added
- [x] Javadoc added or updated
- [ ] Updated the documentation in [docs folder](../docs)

### CODE REVIEW

File: symphony-bdk-spring/bdk-app-starter/pom.xml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (symphony-youri) [inline symphony-bdk-spring/bdk-app-starter/pom.xml:1]: should it be used by the example then?

File: symphony-bdk-examples/bdk-spring-boot-example/src/main/resources/application.yaml [hunk 2]
```diff
@@ -2,14 +2,21 @@ domain: devx1.symphony.com
 
 bdk:
   pod:
+    scheme: https
```
[the remark this answers is not in the document]
Comment (symphony-thibault) [reply symphony-bdk-examples/bdk-spring-boot-example/src/main/resources/application.yaml:5]: Yes It should be the case, I opened [APP-3072](https://perzoinc.atlassian.net/browse/APP-3072) to fix that

File: symphony-bdk-spring/bdk-core-starter/pom.xml [hunk 3]
```diff
@@ -49,12 +53,72 @@
...
+                <artifactId>maven-surefire-plugin</artifactId>
+            </plugin>
+            <plugin>
+                <groupId>org.jacoco</groupId>
+                <artifactId>jacoco-maven-plugin</artifactId>
+                <configuration>
+                    <excludes>
+                        <!-- Too difficult to cover, I'll have a look later -->
+                        <exclude>com/symphony/bdk/spring/service/DatafeedAsyncLauncherService.class</exclude>
+                        <exclude>com/symphony/bdk/spring/config/BdkActivityConfig$SlashAnnotationProcessor.class</exclude>
```
[the remark this answers is not in the document]
Comment (symphony-thibault) [reply symphony-bdk-spring/bdk-core-starter/pom.xml:95]: But I cannot reach `90%` :(

[the remark this answers is not in the document]
Comment (symphony-thibault) [reply symphony-bdk-spring/bdk-core-starter/pom.xml:95]: Tested but I cannot reach `90%` of code coverage

File: symphony-bdk-spring/bdk-app-starter/pom.xml (hunk 1 above)
Comment (symphony-thibault) [reply symphony-bdk-spring/bdk-app-starter/pom.xml:1]: Just created the module, no implementation provided yet

**Review by symphony-elias — APPROVED** [review]
LGTM

**Pull request merged**